packages/react-form/README.md

```diff
@@ -698,88 +731,65 @@ You can choose to initialize the list with an existing number of cards or no car
-  cvv: '',
-});
+      <Button
+        disabled={index === fields.length - 1}
+        onClick={() => moveItem(index, index + 1)}
+      >
+        Move Item Down
+      </Button>
+      <hr />
+    </div>
+  ))}
+  <Button onClick={() => addItem()}>Add Card</Button>
+  <Button disabled={!dirty} onClick={reset}>
+    Reset
+  </Button>
+</FormLayout>
 ```
 
-You can then use this argument to do as you wish :).
-
-#### Does this work with useForm?
+##### How to use it with `useForm`
 
-Yes this works out of the box with `useForm` and is compatible with what `useList` is compatible with.
+You can use `useDynamicList` with `useForm` in two ways:
 
-You would utilize it the following way in order to use the forms reset and dirty state for the `DynamicList` as well as others fields.
+You can directly pass one or several dynamic lists within `fields`. But note that the `reset` function returned by `useForm` won't reset the dynamic list to its initial value. It will only reset the value of each field for each item. So for example if you remove an item, clicking on reset won't re-add it. Plus the adding or removing a field won't make the form dirty. The form will only be dirty if you edit the value of one item's field.
 
 ```tsx
-const {submit, dirty, submitting, reset} = useForm({
+const form = useForm({
   fields: {
-    title: useField('')
-  },
-  dynamicLists: {
-    customerCards: useDynamicList<Card>([{cardNumber: '4242 4242 4242 4242', cvv: '422'}], emptyCardFactory)
+    customerCards: fields, // the fields destructured from useDynamicList
   },
   onSubmit: async fieldValues => {
     console.log(fieldValues);
-    return {status: 'success'};
+    return submitSuccess();
   },
 });
 ```
 
-you can directly pass a dynamic list within `fields` but note that the `reset` function returned by `useForm` will only reset each field, plus the `dirty` state will only reflect dirty fields within each item.
+Or you can pass your dynamic lists within `dynamicLists`. This way the form `reset` and `dirty` will work as expected. Note that you still need to have one "classic field", `fields` cannot be optionnal at the moment. If you use TypeScript, you might want to pass generic types to `useForm` in order to make sure TypeScript will infer the `fieldValues` types correctly within `onSubmit`.
 
 ```tsx
-const {submit, dirty, submitting, reset} = useForm({
-  fields: {
-    fields,  // the fields returned from useDynamicList.
+const fields = {
+  title: useField(''),
+};
+
+const customerCards = useDynamicList<Card>([], emptyCardFactory);
+
+const form = useForm<typeof fields, typeof customerCards>({
+  fields,
+  dynamicLists: {
+    customerCards,
   },
   onSubmit: async fieldValues => {
     console.log(fieldValues);
-    return {status: 'success'};
+    return submitSuccess();
   },
 });
 ```
 
-A code sandbox of `useDynamicList` in action can be seen [here](https://codesandbox.io/s/hungry-rubin-exrkz?fontsize=14&hidenavigation=1&theme=dark)
```

Comment:
I am not sure we should remove the sandbox as it helps people visualize how the dynamic list works

Reply:
If you think something i missing in the README then we can add it tho